File: SuperiorPackGroup/BLL/BomConsumptionBLL.vb
Imports DXDAL.SPGData
Imports DevExpress.Data.Filtering
Imports DevExpress.Xpo

Public Class BomConsumptionBLL

    Public Shared Function UpdateBomConsumption(ByVal mySession As Session, ByVal rawMatID As Items, ByVal inventoryID As Inventory, ByVal quantityUsed As Single, ByVal scrapFactorUsed As Single) As Single

        Dim bom As BomConsumption
        Dim originalQuantity As Single = 0
        Dim originalScrap As Single = 0
        Dim consumptionCriteria As CriteriaOperatorCollection = New CriteriaOperatorCollection()

        consumptionCriteria.Add(New BinaryOperator(BomConsumption.Fields.RawMatID.ItemID.PropertyName, rawMatID.ItemID, BinaryOperatorType.Equal))
        consumptionCriteria.Add(New BinaryOperator(BomConsumption.Fields.InventoryID.InventoryID.PropertyName, inventoryID.InventoryID, BinaryOperatorType.Equal))

        bom = mySession.GetObjectByKey(Of BomConsumption)(mySession.Evaluate(Of BomConsumption)(CriteriaOperator.Parse("Oid"), New GroupOperator(GroupOperatorType.And, consumptionCriteria)))

        If bom Is Nothing Then
            bom = New BomConsumption(mySession)
            bom.RawMatID = rawMatID
            bom.InventoryID = inventoryID
        Else
            originalQuantity = bom.QuantityUsed
            originalScrap = CSng(bom.ScrapFactorUsed)
        End If

        bom.QuantityUsed = quantityUsed
        bom.ScrapFactorUsed = scrapFactorUsed

        bom.Save()

        Return CType(bom.QuantityUsed - originalQuantity + bom.ScrapFactorUsed - originalScrap, Single)

    End Function

End Class

File: SuperiorPackGroup/BLL/BomConsumptionBLL.cs
//INSTANT C# NOTE: Formerly VB project-level imports:
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;

using DXDAL.SPGData;
using DevExpress.Data.Filtering;
using DevExpress.Xpo;

namespace SuperiorPackGroup
{
	public class BomConsumptionBLL
	{

		public static float UpdateBomConsumption(Session mySession, Items rawMatID, Inventory inventoryID, float quantityUsed, float scrapFactorUsed)
		{

			BomConsumption bom = null;
			float originalQuantity = 0;
			float originalScrap = 0;
			CriteriaOperatorCollection consumptionCriteria = new CriteriaOperatorCollection();

			consumptionCriteria.Add(new BinaryOperator(BomConsumption.Fields.RawMatID.ItemID.PropertyName, rawMatID.ItemID, BinaryOperatorType.Equal));
			consumptionCriteria.Add(new BinaryOperator(BomConsumption.Fields.InventoryID.InventoryID.PropertyName, inventoryID.InventoryID, BinaryOperatorType.Equal));

			bom = mySession.GetObjectByKey<BomConsumption>(mySession.Evaluate<BomConsumption>(CriteriaOperator.Parse("Oid"), new GroupOperator(GroupOperatorType.And, consumptionCriteria)));

			if (bom == null)
			{
				bom = new BomConsumption(mySession);
				bom.RawMatID = rawMatID;
				bom.InventoryID = inventoryID;
			}
			else
			{
				originalQuantity = bom.QuantityUsed;
				originalScrap = (float)bom.ScrapFactorUsed;
			}

			bom.QuantityUsed = quantityUsed;
			bom.ScrapFactorUsed = scrapFactorUsed;

			bom.Save();

			return (float)(bom.QuantityUsed - originalQuantity + bom.ScrapFactorUsed - originalScrap);

		}

	}

}